modules.d/90dmsquash-live/dmsquash-live-root.sh

```diff
@@ -33,19 +33,34 @@ overlay_size=$(getarg rd.live.overlay.size=)
 getargbool 0 rd.live.overlay.thin && thin_snapshot="yes"
 getargbool 0 rd.live.overlay.overlayfs && overlayfs="yes"
 
+# Take a path to a disk label and return the patent disk if it is a partition
+# Otherwise returns the original path
+function get_check_dev() {
+    dev_path="$(udevadm info -q path --name "$1")"
+    udevadm info -q property --path "${dev_path}" | grep -q 'DEVTYPE=partition' || { echo "$1"; return; }
+    parent="${dev_path%/*}"
+    udevadm info -q property --path "${parent}" | grep -q 'DEVTYPE=disk' || { echo "$1"; return; }
```

Comment:
Probably best to save this to a variable and use strstr as well.

Also, best to check with ^ and $ on strings with grep (not sure about strstr).

Reply:
You can't use ^$, the output ends up as one big string so you are actually testing to see if it is present. Good idea saving it to a variable though.